scipy/sparse/_coo.py

```diff
@@ -304,7 +305,7 @@ def tocsc(self, copy=False):
         else:
             M,N = self.shape
             idx_dtype = self._get_index_dtype(
-                (self.col, self.row), maxval=max(self.nnz, M)
+                (self.col, self.row), maxval=max(self.nnz, *self.shape)
```

Comment:
Why was this change needed?

Reply:
If self.shape is e.g. `(5, 2**33)` and no nonzero values have index higher than `(2**32 -1)` then the current code will make `idx_type` 32-bit while it needs to be 64-bit. right?  
The current code takes into account large number of rows, but not large number of columns.

I think this is a bug in the 2d code. I could make it a separate PR if that is better. And I guess it should have a test which I haven't put into this either.  Would this be better as a separate PR?